### ISSUE TITLE
.NET 3.5 Version 8.9.15 requires external PhoneNumberMetaData.xml

### ISSUE THREAD

**Issue by athoik** [twcclegg/libphonenumber-csharp#78]
Hi, 

I just updated an old project to a newer version 8.9.15 from 7.5.1.

Currently the PhoneNumberUtil.GetInstance() fails with error `Could not find file 'C:\...\bin\Debug\PhoneNumbers.PhoneNumberMetaData.xml'`.

So it's now mandatory to place the [PhoneNumberMetaData.xml](https://github.com/googlei18n/libphonenumber/blob/master/resources/PhoneNumberMetadata.xml) in the output folder?

I though that metadata was embedded in the dll.

Maybe [example](https://github.com/twcclegg/libphonenumber-csharp#example) should be updated if xml is not embedded any more?

Thanks!

**Comment by twcclegg** [comment]
Metadata is embedded for other versions. .NET 3.5 support was briefly removed to support netstandard. I will take a look.

**Comment by twcclegg** [comment]
Are you having this issue with the nuget package or using the source code? It is published as Release rather than Debug.

**Comment by athoik** [comment]
I am using the nuget package.

**Comment by twcclegg** [comment]
@neuroid don't suppose you have seen this?

**Comment by neuroid** [comment]
Nope. But, I'm still on 8.9.3 in my project. I'll check the latest version tomorrow.

**Comment by athoik** [comment]
8.9.8 is the first version that issue appears.

**Comment by twcclegg** [comment]
@athoik Can you try https://ci.appveyor.com/api/buildjobs/hfoou3xgdxyrdbjt/artifacts/csharp%2FPhoneNumbers%2Fbin%2FRelease%2Flibphonenumber-csharp.8.9.15-040b0d0.nupkg?

**Comment by athoik** [comment]
It doesn't fix the issue, still getting the same error when creating instance.

**Comment by athoik** [comment]
@twcclegg, I just a quick look on the code, currently if NET35 we are not trying to load embedded XML at all.

Maybe something like that will do it? 

```
--- a/csharp/PhoneNumbers/BuildMetadataFromXml.cs
+++ b/csharp/PhoneNumbers/BuildMetadataFromXml.cs
@@ -80,10 +80,7 @@ namespace PhoneNumbers
             bool isAlternateFormatsMetadata)
         {
             XDocument document;
-#if NET35
-            document = XDocument.Load(name);
-#else
-#if  NET40
+#if (NET35 || NET40)
             var asm = Assembly.GetExecutingAssembly();
 #else
             var asm = typeof(PhoneNumberUtil).GetTypeInfo().Assembly;
```

**Comment by athoik** [comment]
@twcclegg , I created a PR #80. I'll check the result from CI and let you know.

**Comment by neuroid** [comment]
Sorry for the delay but I finally managed to test this. 8.9.7 is the last version that is working for me. Looking at the changes in 8.9.8, @athoik's PR seems to be a step in the right direction (although, the CI build failed so this needs more work).

**Comment by neuroid** [comment]
@athoik, could you try this patch?

```diff
diff --git a/csharp/PhoneNumbers/BuildMetadataFromXml.cs b/csharp/PhoneNumbers/BuildMetadataFromXml.cs
index f4e8a82a..69dcbf69 100644
--- a/csharp/PhoneNumbers/BuildMetadataFromXml.cs
+++ b/csharp/PhoneNumbers/BuildMetadataFromXml.cs
@@ -20,6 +20,9 @@ using System.IO;
 using System.Linq;
 using System.Reflection;
 using System.Text.RegularExpressions;
+#if NET35
+using System.Xml;
+#endif
 using System.Xml.Linq;

 namespace PhoneNumbers
@@ -80,19 +83,20 @@ namespace PhoneNumbers
             bool isAlternateFormatsMetadata)
         {
             XDocument document;
-#if NET35
-            document = XDocument.Load(name);
-#else
-#if  NET40
+
+#if  NET40 || NET35
             var asm = Assembly.GetExecutingAssembly();
 #else
             var asm = typeof(PhoneNumberUtil).GetTypeInfo().Assembly;
 #endif
             using (var input = asm.GetManifestResourceStream(name))
-            {
+            {
+#if NET35
+                document = XDocument.Load(new XmlTextReader(input));
+#else
                 document = XDocument.Load(input);
-            }
 #endif
+            }

             var metadataCollection = new PhoneMetadataCollection.Builder();
             var metadataFilter = GetMetadataFilter(liteBuild, specialBuild);
```

This fixes the issue for me and compiles for all runtimes.

**Comment by athoik** [comment]
Hi @neuroid,

Yes, your approach seems right, the XmlTextReader supports stream input.

Let's see if we finally made it!

**Comment by athoik** [comment]
Yes,  we did 👍

CI build worked, testing the produced dll and there is no exception any more on .NET 3.5

In case you like to commit my trial and error attempts, please consider "squash and merge".

Thanks!

**Comment by twcclegg** [comment]
See where things got [broken](https://github.com/twcclegg/libphonenumber-csharp/compare/v8.9.7...7e27cadae86f0ef10e0c04c8eef5c654730b9b66#diff-f71d0c18334f226d9488d3114c43175bL725) I'll merge #80 and release it with 8.9.16, date for that depends on Google's release but I would guess the next couple days

**Comment by twcclegg** [comment]
Fix release in v8.9.16.
